std/bitmanip.d

```diff
@@ -1668,25 +1668,29 @@ public:
     Throws:
       `Exception` if the given string contains invalid characters.
     */
-    static BitArray fromString(string s) pure @system
+    static BitArray fromString(S)(S s) pure @system
+    if (isSomeString!S)
     {
-        import std.exception : enforce;
-        import std.string : lastIndexOfNeither;
         import std.algorithm.searching : count;
 
-        enforce(s.lastIndexOfNeither("01_") == -1,
-            "String contains invalid characters, only '0', '1', '_' are allowed.");
-
         BitArray ba;
         ba.length = s.length - s.count('_');
         size_t i = 0;
         foreach (c; s)
         {
-            if (c == '_')
-                continue;
-            if (c == '1')
-                ba[i] = 1;
-            i++;
+            switch (c)
+            {
+                case '1':
+                    ba[i] = 1;
+                    goto case;
```

Comment:
why not `break;`?

Reply:
In both cases, `0` and `1`, I need to increment `i`.

Reply:
`ba[i++]` - this makes the intent clear to the reader, `goto` should only be used if absolutely necessary.

Reply:
then shouldn't it be` goto case '0';`?

Reply:
`goto case;` is to mark an intended fallthrough, i.e. goes to the next case. I think I'll start from `i = -1` and then use `ba[i++] = 1`, as you say (in D `i++` is exactly the same as `++i`, right?).

Reply:
Nope - why would we have them then:

https://dlang.org/spec/operatoroverloading.html#postincrement_postdecrement_operators

Reply:
For some reason I thought it was changed in D. Maybe because you overload both of them at the same time. Thanks for pointing this out.